iroh-relay/src/server/client.rs

```diff
@@ -365,7 +365,9 @@ impl Actor {
         match frame {
             Frame::SendPacket { dst_key, packet } => {
                 let packet_len = packet.len();
-                self.handle_frame_send_packet(dst_key, packet)?;
+                if let Err(err) = self.handle_frame_send_packet(dst_key, packet) {
```

Comment:
Hum, I guess you somehow have to match the type here for your custom type to add protection against the type changing?  The `@` syntax in match might be able to do that, not sure if you can use that in `if let`.

Reply:
Good point. This seems to work, cool:
```rust
                if let Err(err @ ForwardPacketError { .. }) =
                    self.handle_frame_send_packet(dst_key, packet)
                {
                    warn!("failed to handle send packet frame: {err:#}");
                }

```

Reply:
Nice!  Good to know if-let also allows the @ syntax.